### ISSUE TITLE
Receiver is unable to find message size despite parsing correctly

### ISSUE THREAD

**Issue by M0n7y5** [chronoxor/FastBinaryEncoding#79]
Hi, for some reason when i send some message from my server to client, client will receive it, parse it and invoke OnReceive(myType msg) correctly. After going out of scope, it throws assert error.  
![obrazek](https://user-images.githubusercontent.com/17201053/200202710-bec74154-f290-4a56-a14a-72b056ef8c20.png)

After some investigation. i found out it that it tries to get message size like this:
![obrazek](https://user-images.githubusercontent.com/17201053/200202761-758bed66-e604-4624-9686-a6e9e862cefb.png)

offset2 is 58 in this case:
![obrazek](https://user-images.githubusercontent.com/17201053/200202822-7d4b60e6-0f76-461d-8505-187db9c25fe0.png)

resulted message size is 0;

This is the content of the buffer
![obrazek](https://user-images.githubusercontent.com/17201053/200202855-4fce2936-3606-48a1-98c1-11495310bb06.png)

This is called only once:
![obrazek](https://user-images.githubusercontent.com/17201053/200202956-3a34a2e1-4851-4e0e-bc20-75455b54a88b.png)

![obrazek](https://user-images.githubusercontent.com/17201053/200203169-8a150297-e1ca-4408-8d48-4828e98acf61.png)

For some reason it tries to invalidate this message after invoking OnReceive(myType msg) and data inside that struct are correct. So i don't quite understand the callgraph here. Here is callstack:
![obrazek](https://user-images.githubusercontent.com/17201053/200203132-da387705-0dfd-48fe-845f-78a00e9c958a.png)

**Comment by chronoxor** [comment]
First of all please check that you call Reset() for each time, you client is connected:
```
        void OnConnected()
        {
            // Reset FBE protocol buffers
            Reset();
            ....
        }
```

**Comment by chronoxor** [comment]
If you have your own Server & Client implementation, I suggest starting from ProtoServer/ProtoClient examples to get everything works fine and replace TcpServer/TcpSession/TcpClient with your own transport layer.

**Comment by M0n7y5** [comment]
> If you have your own Server & Client implementation, I suggest starting from ProtoServer/ProtoClient examples to get everything works fine and replace TcpServer/TcpSession/TcpClient with your own transport layer.

That's what i am trying to do. But the documentation here is outdated and the fact that i need to go to completely different repository in order to get proper info about how to use this project is kinda red flag for me...
For transport i am using Asp.Net websockets with custom encryption. I tested it all before i tried to plug in FBE for some actual protocol. It works flawlessly. But now am getting confusing crashes when i am trying to use FBE;

> First of all please check that you call Reset() for each time, you client is connected:
> 
> ```
>         void OnConnected()
>         {
>             // Reset FBE protocol buffers
>             Reset();
>             ....
>         }
> ```
I am already doing that.

Fun fact... Setting this to 0 here makes it work without crash but it is printing this infinitely
![obrazek](https://user-images.githubusercontent.com/17201053/200206590-c40b0f4b-b177-4e92-9eea-532ac8f2de8a.png)

![remedybg_Avwa9Y8MNR](https://user-images.githubusercontent.com/17201053/200206601-44b842cd-3d6c-422b-b88e-c84b41018338.png)

So that means it correctly get message size. But still why is it bale to print that message and the after printing its get validated?
Shouldn't be this other way? Invalidate message -> Parse -> Invoke handlers?

**Comment by M0n7y5** [comment]
I found the issue... The issue was additional message byte padding created by encryption algorithm. Since FBE tries to parse entire provided buffer. It was parsing first message correctly and then it tried to parse rest of the buffer. Since excess bytes are null, static assert with invalid message size is triggered. I made some additional changes and now it seems to be working fine. 
I was under the impression that FBE will ignore those padding bytes which DOES but not in Debug mode.
